platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/style/BackgroundLayerTest.java

```diff
@@ -55,189 +58,249 @@ protected Class getActivityClass() {
 
   private void setupLayer(){
     Timber.i("Retrieving layer");
-    layer = mapboxMap.getLayerAs("background");
+    invoke(mapboxMap, new MapboxMapAction.OnInvokeActionListener() {
+      @Override
+      public void onInvokeAction(UiController uiController, MapboxMap mapboxMap) {
+        layer = mapboxMap.getLayerAs("background");
+      }
+    });
   }
 
   @Test
   public void testSetVisibility() {
     validateTestSetup();
     setupLayer();
     Timber.i("Visibility");
-    assertNotNull(layer);
-
-    // Get initial
-    assertEquals(layer.getVisibility().getValue(), VISIBLE);
-
-    // Set
-    layer.setProperties(visibility(NONE));
-    assertEquals(layer.getVisibility().getValue(), NONE);
+    invoke(mapboxMap, new MapboxMapAction.OnInvokeActionListener() {
+      @Override
+      public void onInvokeAction(UiController uiController, MapboxMap mapboxMap) {
+        assertNotNull(layer);
+
+        // Get initial
```

Comment:
Same thing happens in `CircleLayerTest.java`, `FillExtrusionLayerTest.java`, `FillLayerTest.java`, `LineLayerTest.java`, `RasterLayerTest.java` (generated classes).